controllers/storagecluster/cephblockpools.go

```diff
@@ -49,7 +49,6 @@ func (r *StorageClusterReconciler) newCephBlockPoolInstances(initData *ocsv1.Sto
 		mirroringPeerSpec, err := r.addPeerSecretsToCephBlockPool(initData)
 		if err != nil {
 			r.Log.Error(err, "Unable to add cluster peer token to CephBlockPool.", "CephBlockPool", klog.KRef(poolNamespace, poolName))
-			return nil, err
```

Comment:
To fully fill mirroring, we definitely need peer cluster secret reference here. But is it ok to fail StorageCluster CR? it blocks a lot right. I thought storage cluster resource reconcile is the main. 

So I thought instead of failing to reconcile better we raise error log message to the user like peer secret name is empty or invalid.

But yes without secret reference mirroring is not possible

Reply:
Everything in this directory is part of the StorageCluster reconciliation. The StorageCluster Spec dictates what the admin's desired state is, and as part of that (by default) we observe and update a CephBlockPool as needed. In addition, not having a valid CephBlockPool means that several other items down the line will be broken as well. That said, if the block pool itself *still works*, and it just can't do mirroring, then I think it's fine to not fail here.

Reply:
ack, Thanks for the explanation and it completely makesense. Mirroring is additional and it affected other cephBlockPool behaviors which are unrelated to mirroring.

Reply:
:+1: